### ISSUE TITLE
WeightNorm: Add reset_parameters Linear override

### ISSUE THREAD

**Issue by a-canela** [pytorch/pytorch#76636]
### 🚀 The feature, motivation and pitch

I have faced some problems when trying to **reset the parameters of a Linear module to which I have applied weight normalization**. I have realized that the `reset_parameters` instance method of the Linear module doesn't consider potential modifications of its parameters attributes names, i.e., `self.weights` might be split into `self.weight_g` and `self.weight_v` when using `weight_norm` for instance, so this instance method implementation become kind of useless in these cases.

### Alternatives

I guess that a proper solution may lead to some refactoring, but a naive solution could be just something like this with any luck:

https://github.com/a-canela/pytorch/pull/1/files

### Additional context

Reproducing the issue:

```
import torch

module = torch.nn.utils.weight_norm(torch.nn.Linear(20, 40), name='weight')
params_before = {k: p.mean() for k, p in module.named_parameters()}
module.reset_parameters()
params_after = {k: p.mean() for k, p in module.named_parameters()}
assert all([params_before[k] != params_after[k] for k in params_before])
```

**Comment by jbschlosser** [comment]
What's the status on refactoring `weight_norm` to use parametrization? cc @Lezcano @albanD I believe that would fix this issue since a proper inverse would be defined, is that correct?

**Comment by Lezcano** [comment]
I don't think anyone is working on it, but it shouldn't be too difficult to implement. I've just never found the time to do it :(
